test/juxt/site/graphql_test.clj

```diff
@@ -459,3 +459,165 @@ type Mutation {
 }
 "
       ))
+
+(def cascade-mutation-test-data
+  "Data for evict-cascade and delete-cascade tests"
+  [[:xtdb.api/put access-all-areas]
+
+   [:xtdb.api/put
+    {:xt/id "https://example.org/graphql"
+     :doc "A GraphQL endpoint"
+     :juxt.http.alpha/methods #{:post :put :options}
+     :juxt.http.alpha/acceptable "application/graphql"
+     :juxt.site.alpha/put-fn 'juxt.site.alpha.graphql/put-handler
+     :juxt.site.alpha/post-fn 'juxt.site.alpha.graphql/post-handler}]
+
+   [:xtdb.api/put
+    {:xt/id "https://example.org/persons/ts1"
+     :type "Person"
+     :name "Testuser 1"}]
+
+   [:xtdb.api/put
+    {:xt/id "https://example.org/persons/ts2"
+     :type "Person"
+     :name "Testuser 2"}]
+
+   [:xtdb.api/put
+    {:xt/id "https://example.org/house/hh1"
+     :type "House"
+     :name "House 1"
+     :ownerId "https://example.org/persons/ts2"
+     }]
+
+   [:xtdb.api/put
+    {:xt/id "https://example.org/house/hh2"
+     :type "House"
+     :name "House 2"
+     :ownerId "https://example.org/persons/ts2"
+     }]
+
+   [:xtdb.api/put
+    {:xt/id "https://example.org/house/hh3"
+     :type "House"
+     :name "House 3"
+     :ownerId "https://example.org/persons/ts1"
+     }]
+   ])
+
+(def cascade-schema
+  "Schema for cascade-evict and cascade-delete tests"
+  "schema {
+             query: Query
+             mutation: Mutation
+           }
+           type Query {
+                 persons: [Person] @site(q: { find: [e] where: [[e {keyword: \"type\"} \"Person\"]]})
```

Comment:
Query definition for persons uses explicit site query directive to specify datalog query for fetching user entities.
According to my understanding of the site (based mostly on the site tutorial), this shouldn't be required, and the site should be able to automatically infer correct datalog queries from schema definition, but it doesn't work, and I was unable to find the root cause of the issue despite many hours spend debugging this.
Can somebody have a quick look at whether I'm doing something obviously wrong in schema definition/seed-data in this test case and why querying persons via site/graphql returns no results unless an explicit datalog query is provided in the schema definition? @armincerf

Reply:
Yeah you shouldn't need an explicit query here, however the default type key is `:juxt.site/type` so the implicit query is 
```
{:find [e] :where [[e :juxt.site/type "Person"]]}
```
which is why it wouldn't work implicitly.

to debug that you could have put some logs in the juxt.site.alpha.graphql/infer-query function but probably the documentation could better reflect how the implicit queries are generated

Reply:
really every schema should have an explicit type directive on the schema decleration though, so you could put @site(type: "type") at the top of the schema and that would also make it work

Reply:
I played a lot with attr namespaces and neither seeding data with `:juxt.site/type` nor adding `@site(type: "type")` to graphql schema definition solves that, still empty result set.
Thanks for the infer query debug suggestions, I tried that and here is the implicit query:
```clj
{:find [e _siteCreatedAt], :where [[e :type "Person"] [e :_siteCreatedAt _siteCreatedAt]]}
```
So the problem seems to be that implicit queries expects `:_siteCreatedAt` attribute to be always present, which is not the case when manually creating seed data directly in tests.
I don't really need it in this PR, but I finally know how to fix it in the future, thanks :)

Reply:
Ah yes I forgot about that, this is why it is recommended to never manually insert things into the db and instead use the `prepare-mutation-entity` or `args-to-entity` functions, though this could definitely be improved, ideas are welcomed 😅